hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -55,7 +50,29 @@ UnwrappedMessageContent unwrapContent(byte[] data, Schema schema) {
         return builder.build();
     }
 
+    private long getTimestamp(Map<Utf8, Utf8> metadata) {
+        return metadata != null && metadata.containsKey(METADATA_TIMESTAMP_KEY) ? timestampFromMetadata(metadata) :
+                System.currentTimeMillis();
```

Comment:
We have special object type of Clock to get current time. Advantage of inject & use it is that you can mock it in unit tests and verify if proper value is set.